platform-sdk/swirlds-benchmarks/src/jmh/java/com/swirlds/benchmark/reconnect/MerkleBenchmarkUtils.java

```diff
@@ -0,0 +1,185 @@
+
+        final MerkleLeaf A = new BenchmarkMerkleLeaf("A");
+        final MerkleInternal i0 = new BenchmarkMerkleInternal("i0");
+        final MerkleInternal i1 = new BenchmarkMerkleInternal("i1");
+        root.setChild(0, A);
+        root.setChild(1, i0);
+        root.setChild(2, i1);
+
+        final MerkleLeaf B = new BenchmarkMerkleLeaf("B");
+        final MerkleLeaf C = new BenchmarkMerkleLeaf("C");
+        i0.setChild(0, B);
+        i0.setChild(1, C);
+
+        final MerkleLeaf D = new BenchmarkMerkleLeaf("D");
+        i1.setChild(0, D);
+        i1.setChild(1, null);
+
+        initializeTreeAfterCopy(root);
+        return root;
+    }
+
+    static <T extends MerkleNode> T hashAndTestSynchronization(
+            final MerkleNode startingTree,
+            final MerkleNode desiredTree,
+            final Configuration configuration,
+            final ReconnectConfig reconnectConfig)
+            throws Exception {
+        System.out.println("------------");
+        System.out.println("starting: " + startingTree);
+        System.out.println("desired: " + desiredTree);
+
+        if (startingTree != null && startingTree.getHash() == null) {
+            MerkleCryptoFactory.getInstance().digestTreeSync(startingTree);
+        }
+        if (desiredTree != null && desiredTree.getHash() == null) {
+            MerkleCryptoFactory.getInstance().digestTreeSync(desiredTree);
+        }
+        return testSynchronization(startingTree, desiredTree, configuration, reconnectConfig);
+    }
+
+    /**
+     * Synchronize two trees and verify that the end result is the expected result.
+     */
+    @SuppressWarnings("unchecked")
+    private static <T extends MerkleNode> T testSynchronization(
+            final MerkleNode startingTree,
+            final MerkleNode desiredTree,
+            final Configuration configuration,
+            final ReconnectConfig reconnectConfig)
+            throws Exception {
+        try (PairedStreams streams = new PairedStreams(configuration.getConfigData(SocketConfig.class))) {
+            final LearningSynchronizer learner;
+            final TeachingSynchronizer teacher;
+
+            learner = new LearningSynchronizer(
+                    getStaticThreadManager(),
+                    streams.getLearnerInput(),
+                    streams.getLearnerOutput(),
+                    startingTree,
+                    () -> {
+                        try {
+                            streams.disconnect();
+                        } catch (final IOException e) {
+                            // test code, no danger
+                            e.printStackTrace();
+                        }
+                    },
+                    reconnectConfig);
+            final PlatformContext platformContext =
+                    BenchmarkPlatformContextBuilder.create().build(configuration);
+            teacher = new TeachingSynchronizer(
+                    platformContext.getConfiguration(),
```

Comment:
I wonder if `configuration` can be used here instead of `platformContext.getConfiguration()`? If so, the whole BenchmarkPlatformContextBuilder class is not needed